vnext/Microsoft.ReactNative/Fabric/Composition/CompositionRootAutomationProvider.cpp

```diff
@@ -1,6 +1,13 @@
 #include "pch.h"
 #include "CompositionRootAutomationProvider.h"
+#include <algorithm>
 #include "UiaHelpers.h"
+#pragma warning(push)
+#pragma warning(disable : 4229)
+#define IN
+#define OUT
+#include <atlsafe.h>
```

Comment:
Please do not use ATLSafe.h

Reply:
this PR adds SAFEARRAY support to WIL which we should be using instead:
https://github.com/microsoft/wil/pull/145

Reply:
+ @carlos-zamora who filed the original wil issue. Can you help? :)

Reply:
Sure! Here's the way we did it in the Terminal repo:
https://github.com/microsoft/terminal/blob/c9e993a38ec04833e15edac230300948a0fab83b/src/cascadia/inc/cppwinrt_utils.h#L222-L250

And here's a quick example of us using it:
https://github.com/microsoft/terminal/blob/c9e993a38ec04833e15edac230300948a0fab83b/src/cascadia/TerminalControl/InteractivityAutomationPeer.cpp#L206

Hope this helps!

Reply:
Nice, I wasn't aware of the WIL API.  @asklar I've been pulled off this work for now.  Would you be OK with me just using the SAFEARRAY directly and opening an issue to move this to the WIL stuff?

Reply:
Wow, so we have to take a whole new dependency just to interface with this?

Reply:
@jonthysell we can interface manually (which I think is what @FalseLobster is proposing) but it's going to be a lot more code than necessary and hard to maintain/reason over. WIL is header only and designed to simplify calling win32 apis so your code becomes a lot more concise. Ideally we can move to that sooner than later.

Reply:
@FalseLobster I'm okay with taking things as you've written them and opening an issue to replace ATL with WIL and assign it to me. There's probably other places the (Fabric) codebase could benefit from access to WIL helpers to deal with all this old Win32 stuff.

@asklar Yes, WIL is header only, but it's also "yet another re-wrapping of standard C++" to add to this project. Like do we really need every implementation of smart pointers that have ever been made? We have stl, boost, Mso, cppwinrt, now wil.

Reply:
the alternative being used here is to use ATL. We should not be using ATL. 
WIL provides convenient wrappers for this specific use case. If STL, Boost, MSO or CppWinRT had RAII wrappers for these types, great. I don't believe they do though.

Reply:
@asklar , @jonthysell , could we avoid dependency on the new library?
SAFEARRAY are not so difficult to use directly.
I would expect that C++/WinRT has all the required wrappers for them.

C++/WinRT is also header-only library, and we know the overall impact on the developer experience.
I would stay away from the header-only libraries since they put a huge pressure on the compiler: such headers must be often compiled again and again for each .cpp that slows down compilation no matter how powerful your computer is.

Reply:
C++/WinRT is for working with WinRT, but these are not WinRT apis, so C++/WinRT doesn't help in this case. Feel free to use SAFEARRAY directly.